app/client/package.json

```
@@ -10,6 +10,28 @@
     "packages/*"
   ],
   "cracoConfig": "craco.dev.config.js",
+  "scripts": {
+    "analyze": "yarn cra-bundle-analyzer",
+    "start": "BROWSER=none EXTEND_ESLINT=true REACT_APP_ENVIRONMENT=DEVELOPMENT REACT_APP_CLIENT_LOG_LEVEL=debug HOST=dev.appsmith.com craco start",
+    "build": "./build.sh",
```

I would like to add type-checking before building 
```suggestion
    "build": "yarn run check-types && ./build.sh",
```

I think we can add a `type-check` call right before the build, because if there is something wrong with the types, then there is no point in doing the build. Also in this way we can remove [this code](https://github.com/appsmithorg/appsmith/pull/21442/files#diff-7600caa85f1c89ff1fcd0b4b73e379816ef5a59cd7e61fdb90feb496e704d16aR111-R113).

---

Build doesn't happen if we the previous three checks we added fail. By adding it as a part of build, we are unnecessarily running jest tests.